diff --git a/node_modules/react-native-neat-date-picker/src/components/Content.tsx b/node_modules/react-native-neat-date-picker/src/components/Content.tsx
index fb163e5..57d0c1e 100644
--- a/node_modules/react-native-neat-date-picker/src/components/Content.tsx
+++ b/node_modules/react-native-neat-date-picker/src/components/Content.tsx
@@ -17,7 +17,7 @@ const Content = ({
   colorOptions, chooseYearFirst,
   daysArray, btnDisabled,
   displayTime, setDisplayTime,
-  output, setOutput
+  output, setOutput,onKeyPressCustom,onShortcutPress,enableShortcut
 }: any) => {
   const [showChangeYearModal, setShowChangeYearModal] = useState(chooseYearFirst || false)
 
@@ -38,6 +38,7 @@ const Content = ({
     : getTranslation('en').weekDays
 
   return (
+    <>
     <View style={[styles.container, { backgroundColor }]}>
       <View style={[styles.header, { backgroundColor: headerColor }]}>
 
@@ -84,9 +85,11 @@ const Content = ({
               selectedDateTextColor,
               selectedDateBackgroundColor
             }}
+            onKeyPressCustom={onKeyPressCustom}
           />
         ))}
       </View>
+    
       <View style={styles.footer}>
         <View style={styles.btn_box}>
           <TouchableOpacity style={styles.btn} onPress={onCancelPress}>
@@ -112,6 +115,26 @@ const Content = ({
         }}
       />
     </View>
+
+    {enableShortcut && 
+    <View style={[styles.container, { marginTop:18,backgroundColor,flexDirection:'row',justifyContent:'space-around',paddingVertical:10 }]}>
+
+    <TouchableOpacity style={{borderColor:weekDaysColor,borderWidth:1,padding:10,borderRadius:10,backgroundColor}} onPress={()=>onShortcutPress('LAST_WEEK')}>
+      <Text style={{fontWeight:500}}>Last Week</Text>
+    </TouchableOpacity>
+
+    <TouchableOpacity style={{borderColor:weekDaysColor,borderWidth:1,padding:10,borderRadius:10,backgroundColor}} onPress={()=>onShortcutPress('THIS_WEEK')}>
+      <Text style={{fontWeight:500}}>This Week</Text>
+    </TouchableOpacity>
+
+    <TouchableOpacity style={{borderColor:weekDaysColor,borderWidth:1,padding:10,borderRadius:10,backgroundColor}} onPress={()=>onShortcutPress('NEXT_WEEK')}>
+      <Text style={{fontWeight:500}}>Next Week</Text>
+    </TouchableOpacity>
+    </View>
+    }
+
+    </>
+
   )
 }
 
diff --git a/node_modules/react-native-neat-date-picker/src/components/Key.tsx b/node_modules/react-native-neat-date-picker/src/components/Key.tsx
index e8422a0..329a04d 100644
--- a/node_modules/react-native-neat-date-picker/src/components/Key.tsx
+++ b/node_modules/react-native-neat-date-picker/src/components/Key.tsx
@@ -27,9 +27,10 @@ type KeyProps = {
     mode: Mode;
     output: Output;
     setOutput: Dispatch<SetStateAction<Output>>;
+    onKeyPressCustom:any
 }
 
-const Key: FC<KeyProps> = ({ colorOptions, Day, mode, output, setOutput }: KeyProps) => {
+const Key: FC<KeyProps> = ({ colorOptions, Day, mode, output, setOutput,onKeyPressCustom }: KeyProps) => {
     const { dateTextColor, backgroundColor, selectedDateTextColor, selectedDateBackgroundColor } = colorOptions
     const singleMode = mode === 'single'
     const rangeMode = mode === 'range'
@@ -45,6 +46,7 @@ const Key: FC<KeyProps> = ({ colorOptions, Day, mode, output, setOutput }: KeyPr
 
         if (singleMode) {
             const newOutPut = { ...output, date: newDate }
+            onKeyPressCustom(newOutPut)
             setOutput(newOutPut)
             return
         }
@@ -53,9 +55,11 @@ const Key: FC<KeyProps> = ({ colorOptions, Day, mode, output, setOutput }: KeyPr
             if (shouldSetStartDate) {
                 const newOutPut = { ...output, startDate: newDate, endDate: null }
                 setOutput(newOutPut)
+                onKeyPressCustom(newOutPut)
             } else {
                 const newOutPut = { ...output, endDate: newDate }
                 setOutput(newOutPut)
+                onKeyPressCustom(newOutPut)
             }
         }
     }
@@ -88,13 +92,13 @@ const Key: FC<KeyProps> = ({ colorOptions, Day, mode, output, setOutput }: KeyPr
          * As you can see, I delete an unnecessary extra conditional.
          */
         if (!output.endDate) {
-            return timeOfThisKey === output.startDate?.getTime()
+            return timeOfThisKey === output?.startDate?.getTime()
                 ? selectedColors
                 : notSelectedColors
         }
 
         const startDate = output.startDate as Date
-        const isThisDayInSelectedRange = timeOfThisKey >= startDate.getTime() && timeOfThisKey <= output.endDate.getTime()
+        const isThisDayInSelectedRange = timeOfThisKey >= startDate?.getTime() && timeOfThisKey <= output.endDate?.getTime()
         return isThisDayInSelectedRange ? selectedColors : notSelectedColors
     }
 
diff --git a/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.d.tsx b/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.d.tsx
index ee30d92..fe93b29 100644
--- a/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.d.tsx
+++ b/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.d.tsx
@@ -142,6 +142,46 @@ export type NeatDatePickerProps = {
    * @required
    */
   onConfirm: (arg: Object) => void;
+  /**
+   * This callback will execute when user presses confirm button.
+   *
+   * this prop passes an argument `output` For 'single' mode, output contains two properties `date`, `dateString`.
+   * As for 'range' mode, it contains four properties `startDate`, `startDateString`, `endDate` and `endDateString`
+   *
+   * @example
+   * #### single mode:
+   *
+   * ```ts
+   * const onConfirm = ({ date: Date, dateString: string }) => {
+   *   console.log(date.getTime())
+   *   console.log(dateString)
+   * }
+   * ```
+   *
+   * #### range mode:
+   *
+   * ```ts
+   * const onConfirm = (output) => {
+   *   const {startDate, startDateString, endDate, endDateString} = output
+   *   console.log(startDate.getTime())
+   *   console.log(startDateString)
+   *   console.log(endDate.getTime())
+   *   console.log(endDateString)
+   * }
+   * ```
+   *
+   * @required
+   */
+  onKeyPressCustom?: (arg: Object) => void;
+
+
+
+  enableShortcut?:boolean;
+
+  /**
+   * Sets the Date range according to the range we select
+   */
+  onShortcutPress:(arg:string)=>void;
   /**
    * Set this prop to a date if you need to set an initial starting date when opening the date picker the first time. Only works with 'range' mode.
    */
diff --git a/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.tsx b/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.tsx
index a71450e..0a619b8 100644
--- a/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.tsx
+++ b/node_modules/react-native-neat-date-picker/src/components/NeatDatePicker.tsx
@@ -18,26 +18,34 @@ I18nManager.allowRTL(false)
 const winY = Dimensions.get('screen').height
 
 const NeatDatePicker = ({
-    colorOptions, dateStringFormat,
+    colorOptions, dateStringFormat='yyyy-mm-dd',
     endDate, initialDate,
     isVisible, language,
     maxDate, minDate,
-    modalStyles, mode,
+    modalStyles={ justifyContent: 'center' }, mode,
     onBackButtonPress, onBackdropPress,
     onCancel, onConfirm,
     startDate,
     chooseYearFirst,
-    withoutModal
+    withoutModal,
+    onKeyPressCustom,
+    onShortcutPress,
+    enableShortcut
 }: NeatDatePickerProps) => {
     
+    
     // displayTime defines which month is going to be shown onto the screen
     // For 'single' mode, displayTime is also the initial selected date when opening DatePicker at the first time.
-    const [displayTime, setDisplayTime] = useState(initialDate ?? new Date())
+    const [displayTime, setDisplayTime] = useState(initialDate || startDate || new Date())
+
+
     const year = displayTime.getFullYear()
     const month = displayTime.getMonth()// 0-base
     const date = displayTime.getDate()
     const TODAY = new Date(year, month, date)
 
+    
+
     // output decides which date should be active.
     const [output, setOutput] = useState<Output>(
         mode === 'single'
@@ -48,6 +56,12 @@ const NeatDatePicker = ({
     // If user presses cancel, reset 'output' state to this 'originalOutput'
     const [originalOutput, setOriginalOutput] = useState(output)
 
+
+    useEffect(()=>{
+        setOutput({...output,startDate:startDate || null,endDate:endDate||null})
+        setDisplayTime(initialDate ?? startDate ?? new Date())
+    },[startDate,endDate])
+
     const minTime = minDate?.getTime()
     const maxTime = maxDate?.getTime()
 
@@ -167,7 +181,7 @@ const NeatDatePicker = ({
                 colorOptions, chooseYearFirst,
                 daysArray, btnDisabled,
                 displayTime, setDisplayTime,
-                output, setOutput
+                output, setOutput,onShortcutPress
             }}
         />
     )
@@ -195,17 +209,13 @@ const NeatDatePicker = ({
                     colorOptions, chooseYearFirst,
                     daysArray, btnDisabled,
                     displayTime, setDisplayTime,
-                    output, setOutput
+                    output, setOutput,onKeyPressCustom,onShortcutPress,enableShortcut
                 }}
             />
         </Modal>
     )
 }
 
-NeatDatePicker.defaultProps = {
-    dateStringFormat: 'yyyy-mm-dd',
-    modalStyles: { justifyContent: 'center' }
-}
 
 export default NeatDatePicker
 
@@ -216,6 +226,7 @@ const styles = StyleSheet.create({
         height: winY,
         alignItems: 'center',
         padding: 0,
-        margin: 0
+        margin: 0,
+        
     }
 })
diff --git a/node_modules/react-native-neat-date-picker/src/hooks/useDaysOfMonth.ts b/node_modules/react-native-neat-date-picker/src/hooks/useDaysOfMonth.ts
index a696fd1..a4552fb 100644
--- a/node_modules/react-native-neat-date-picker/src/hooks/useDaysOfMonth.ts
+++ b/node_modules/react-native-neat-date-picker/src/hooks/useDaysOfMonth.ts
@@ -23,7 +23,6 @@ const useDaysOfMonth = (inputYear: number, inputMonth: number, minTime?: number,
     const [dateArray, setDateArray] = useState<DaysArray[]>([])
 
     const days = new Date(inputYear, inputMonth + 1, 0).getDate()
-
     const firstDay = new Date(inputYear, inputMonth, 1).getDay()
 
     const prevMonthDays = new Date(inputYear, inputMonth, 0).getDate()
